### ISSUE TITLE
Sprint: May 30 → June 20

### ISSUE THREAD

**Issue by RichardLitt** [ipfs/pm#112]
# Sprint May 30th

Previous sprint: https://github.com/ipfs/pm/issues/109

## Sprint Discussions

[Schedule](https://github.com/ipfs/pm#sprint-discussion-schedule)

Please take notes in a separate pad, if you can, and link it here.

Endeavour         | Lead            | Time (PDT - **UTC/Z** - CEST) | Pad
:---------------: | :-------------: | :--------------------------: | :----:
sync              | @RichardLitt    | 9:00PDT **16:00Z** 18:00CEST  | IRC: #ipfs on Freenode
infrastructure    | @lgierth        | 10:30PDT **17:30Z** 19:30CEST | https://public.etherpad-mozilla.org/p/ipfs-may-30-infrastructure
apps on ipfs, js-ipfs, libp2p      | @dignifiedquire, @diasdavid | 11:00PDT **18:00Z** 20:00CEST | https://pad.riseup.net/p/sdfljsalfsadj-ipfs-js-ipfs
go-ipfs           | @whyrusleeping  | 12:00PDT **19:00Z** 21:00CEST | https://public.etherpad-mozilla.org/p/ipfs-may-30-go-ipfs

Please add the Agenda to the Pad before the endeavour sprint starts.

## Sprint Deliverables

- Add your tasks below in a comment, that way we only have people listed who are really in the sprint
- Add links to issues down here. Only add things you can finish this sprint.

**Comment by RichardLitt** [comment]
## Infrastructure

#### Lead: @lgierth

#### Participants

@jbenet
@RichardLitt
@whyrusleeping
@haadcode
@Kubuxu

### Agenda

- js-ipfs
- pages deploys


**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

### Notes

- check with daviddias if webrtc-star can be deployed multiple times


##### After sprint meeting is finished, create the respective action items on the Github sprint issue

## js-ipfs

#### Lead: @diasdavid

#### Participants

@daviddias
@dignifiedquire
@Kubuxu
@haadcode
@RichardLitt
@jbenet
@em-ly
@voxelot

### Agenda


**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

### Notes


##### After sprint meeting is finished, create the respective action items on the Github sprint issue

## go-ipfs

#### Lead: @whyrusleeping

#### Participants

### Agenda


**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

### Notes


##### After sprint meeting is finished, create the respective action items on the Github sprint issue

**Comment by RichardLitt** [comment]
#### To Do

- [x] Finish work on the http-api-spec
  - [x] Small PRs to go-ipfs
- [x] Started discussion about 78char limit https://github.com/ipfs/go-ipfs/issues/2783
  - [x] PR go-ipfs to keep descriptions under 78 ruler limit: ipfs/go-ipfs#2782, ipfs/go-ipfs#2780, ipfs/go-ipfs#2779
- [x] Work on standard-readme
- [~] Build badge to check issue lists https://github.com/ipfs/pm/issues/111
  - [x] Started work at https://github.com/RichardLitt/github-task-status
- [x] Attend ScotlandJS

**Comment by Kubuxu** [comment]
### @Kubuxu Mid-sprint update:
* [ ] automagical synopsis generator and no- prefixed options - https://github.com/ipfs/go-ipfs/pull/2785 - would love to hear your comments on https://github.com/ipfs/go-ipfs/issues/2799
* [x] `ipfs version --all` - https://github.com/ipfs/go-ipfs/pull/2790
* [x] setup FUSE workspace (KVM, shared directories, shared daemon)
* [ ] Add CORS headers to Read Only Gateway Default config - https://github.com/ipfs/go-ipfs/pull/2778
* [x] Add check for HEAD response to be empty - https://github.com/ipfs/go-ipfs/pull/2775
* [x] Fix commands --flags not showing flags via HTTP API - https://github.com/ipfs/go-ipfs/pull/2773

**Comment by noffle** [comment]
## noffle mid-sprint

I was largely blocked on my ipfs.files.add work for js-ipfs this week -- david was understandably pretty busy at Web Rebels -- so I focused on code reviews and also gave ipget some attention: notably integrating gx, spinning off a helper gx-go-bootstrap CLI for gx-go projects. Just a few more things before a v0.2.0 release is ready! I also bumped go-ipfs-dep to 0.4.2.

- interface-ipfs-core / js-ipfs
  - [X] CRing @diasdavid's PRs
  - [X] revisions to ipfs.files.add PR
  - [X] catching up with all the work david did over the weekend :)
- ipget
  - [X] added gx to subset of deps to deduplicate diff dep versions causing panic
  - [X] wrote gx-go-bootstrap for providing gx/go projects with a nice go-ipfs-like Makefile for non-gx users to build and install (https://github.com/noffle/gx-go-bootstrap)
  - [X] updated README with building from source and where to get binaries
  - [X] some finangalling to get travis passing again
  - [ ] working toward a v0.2.0 release onto dist.ipfs.io
- go-ipfs-dep
  - [X] bumped to go-ipfs 0.4.2

**Comment by diasdavid** [comment]
### > **Hi everyone, we are postponing the sprint Hangouts a week, from Jun 13 to Jun 20.**

A lot has happened in the past 3 weeks, the IPFS community has pushed a great amount of updates in order to get everything prepared for the participation on the [Decentralised Web Summit](http://www.decentralizedweb.net/) (which was a great success! A lot of updates coming soon) and now we are collecting all the feedback and align our development plans accordingly.

We can't wait to share the updates with everyone (expect blog posts soon) and get back to the normal sprint pace, starting next week! 

Thank you :)

**Comment by Kubuxu** [comment]
@Kubuxu sprint update
----------
In past 3 weeks a lot has happened, here is small summary on my part: I've migrated a lot of depenencies from Godeps to gx -40K LOC from go-ipfs repo. I also fixed few bugs in tests which were resulting in failures 20% of a time (If someone sees random test failure now, report it as separate issue and ping me on GH). I've fixed multiple for smaller bugs and issues, and worked on synopsis generator which needs just a review now.

**Comment by RichardLitt** [comment]
@RichardLitt Sprint update 

I was on vacation for a lot of this sprint; my main things done involve publishing a blogpost on ScotlandJS, starting work standardizing all of the readmes, creating and starting github.com/RichardLitt/github-task-status, and generally PRing http-api-spec and go-ipfs.

**Comment by nginnever** [comment]
@voxelot Update

I got to hang out and spend my birthday in Seattle and at the Internet Archive in SF. It was really great to see everyone and be able to spend time brainstorming new ideas, learning the finer points of IPLD in workshop, and learning whole lot about libp2p (still much data to brain). Put in some work on js-ipfs over the past few weeks (cat command and add command http-api). Also began writing a distributed collaborative text editor with js-ipfs. More on that later this week and will probably ping noffle and haad for collab if interested after I finish setting up the project more. Cheers!

**Comment by nicola** [comment]
Went to the decentralized web summit and hang out with most of you here.
Juan explained to me Filecoin/Cube. We had conversations about the IPLD roadmap https://github.com/ipfs/specs/issues/112, the concept of Transforms and the concept of IPC (inter-planetary computation!), reverse linking - or family links.

**Comment by diasdavid** [comment]
### Sprint Update

- Participation in
  - Blue Yard event (Berlin)
  - WebRebels (Oslo)
  - Decentralized WebSummit (SF)
  - NodeConf (California)
- Fix a bunch of issues for interop
- Refactored:
  - interface-transport
  - interface-connection
  - libp2p-tcp
  - libp2p-websockets
    -- discovered and fixed a bunch of bugs in the process.
- Wrote a comprehensive captain.log for js-ipfs

**Comment by whyrusleeping** [comment]
Went to decentralized websummit and nodeconf, talked to a bunch of people and was amazed at how much everyone loves us. (i'm starting to get suspicious about the fact i never hear bad things)

- debugged and fixed some dht hangs caused by misbehaved remote clients
- did some gx work (gx view fixes, merged gx set, and misc other issues)
- debugged js-go interop issues with @diasdavid 
- disabled utp being on by default
- wrote a detailed captains log update
- misc other stuff
